File: SDKs/pocketsmith.com/2.0/python/src/sdk/models/operations/get_users_id_transaction_accounts.py
from dataclasses import dataclass, field
from typing import List,Optional
from sdk.models import shared


@dataclass
class GetUsersIDTransactionAccountsPathParams:
    id: int = field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetUsersIDTransactionAccountsRequest:
    path_params: GetUsersIDTransactionAccountsPathParams = field()
    

@dataclass
class GetUsersIDTransactionAccountsResponse:
    content_type: str = field()
    status_code: int = field()
    error: Optional[shared.Error] = field(default=None)
    transaction_accounts: Optional[List[shared.TransactionAccount]] = field(default=None)
    

File: SDKs/pocketsmith.com/2.0/typescript/src/sdk/models/operations/getusersidtransactionaccounts.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetUsersIdTransactionAccountsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: number;
}


export class GetUsersIdTransactionAccountsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetUsersIdTransactionAccountsPathParams;
}


export class GetUsersIdTransactionAccountsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.Error;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata({ elemType: shared.TransactionAccount })
  transactionAccounts?: shared.TransactionAccount[];
}
